### ISSUE TITLE
Support RollbackOnFailure when stacks are created

### ISSUE THREAD

**Issue by vjindalradialtheory** [aws/aws-cdk#2562]
How can we specify Rollback on failure false ?

see : https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-add-tags.html

**Comment by ozbillwang** [comment]
Looking for the same, but no luck currently

**Comment by jacques-** [comment]
Also need this feature please

**Comment by EthanGao-oss** [comment]
also needed. some resource creation/deletion failed  but cdk continues executing.

**Comment by SomayaB** [comment]
Thanks for expressing your interest everyone! Someone will update this issue when there is movement.

**Comment by piyushwadhwani** [comment]
must have feature

**Comment by KitFristo** [comment]
This would be a nice feature. Our team has had a plethora of difficulties trying to work with AWS Support on tickets because we cannot disable rollback.

**Comment by eladb** [comment]
Sounds like this is something we should add at the CLI level and also at the framework level as a stack option.

@shivlaks can you pick this up?
@rix0rrr should also be supported in CDK Pipelines

**Comment by mrpackethead** [comment]
This is currrently causing me pain with nested stacks , i can't see whats causing the error, becusae its detleted before i get too look.

**Comment by machulav** [comment]
Please correct me if I'm wrong, but it seems to me that this issue can not be implemented at the moment as [AWS::CloudFormation::Stack](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-stack.html#cfn-cloudformation-stack-notificationarns) currently doesn't support rollback property.

Though, it is supported in [create-stack](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/create-stack.html) CLI command, but as I understand, aws-cdk doesn't use CLI for generating the stack and uses CloudFormation template instead.

**Comment by SarahClough** [comment]
> This is currrently causing me pain with nested stacks , i can't see whats causing the error, becusae its detleted before i get too look.

In CloudFormation, filter by 'deleted stacks' and you should still be able to view the error in the event log.

**Comment by tvb** [comment]
This would be nice to have!

**Comment by abaror** [comment]
hard to debug nested stacks without this...

**Comment by aadhikary** [comment]
Any ETA on this. This seems to be a pain. We use the nested stacks on CDK to speed up the build (which has worked). But during development time this causes a lot of pain in case of failures. Yes you can look at nested deleted stacks to see what caused the error but that still does not allow us to test the whole process in speedy way. A way to continuously update the stack will help build on AWS services much faster.

**Comment by vihangshah8** [comment]
I am currently using cloudformation init and its hard to debug in case of issues since the template rolls back by default. This would be a great feature if added as a flag to the cli itself which can be controlled when we do cdk deploy.

**Comment by ytsipun** [comment]
+1

**Comment by keepler-ivanmarques** [comment]
+1

**Comment by niklaswallerstedt** [comment]
> Thanks for expressing your interest everyone! Someone will update this issue when there is movement.

@SomayaB I ended up here from a google search on how to disable rollback in cdk and couldn't find it right away, saw that this issue and #11103 was still open, but later found  #16307 with the answer.

TL;DR
`cdk deploy -R (--no-rollback)`
